File: src/main/java/com/qsoftware/bubbles/data/DataManager.java
package com.qsoftware.bubbles.data;

import com.qsoftware.bubbles.common.entity.Entity;
import org.bson.BsonDocument;
import org.bson.BsonDouble;

import java.awt.geom.Point2D;

public class DataManager {
    public BsonDocument storeEntity(Entity entity) {
        BsonDocument nbt = new BsonDocument();
        nbt.put("position", storePosition(entity.getPos()));
        nbt.put("data", entity.getState());
        return nbt;
    }

    private BsonDocument storePosition(Point2D pos) {
        BsonDocument nbt = new BsonDocument();
        nbt.put("x", new BsonDouble(pos.getX()));
        nbt.put("y", new BsonDouble(pos.getY()));
        return nbt;
    }
}

File: src/main/kotlin/com/qtech/bubbles/data/DataManager.kt
package com.qtech.bubbles.data

import com.qtech.bubbles.entity.Entity
import net.querz.nbt.tag.CompoundTag
import java.awt.geom.Point2D

class DataManager {
    fun storeEntity(entity: Entity): CompoundTag {
        val nbt = CompoundTag()
        nbt.put("position", storePosition(entity.pos))
        nbt.put("data", entity.getState())
        return nbt
    }

    private fun storePosition(pos: Point2D): CompoundTag {
        val nbt = CompoundTag()
        nbt.putDouble("x", pos.x)
        nbt.putDouble("y", pos.y)
        return nbt
    }
}